plugins/woocommerce-blocks/tests/e2e/tests/product-collection/product-collection.block_theme.spec.ts

```diff
@@ -734,6 +730,172 @@ test.describe( 'Product Collection', () => {
 		} );
 	} );
 
+	test.describe( 'Location is recognised', () => {
+		const filterRequest = ( request: Request ) => {
+			const url = request.url();
+			return (
+				url.includes( 'wp/v2/product' ) &&
+				url.includes( 'isProductCollectionBlock=true' )
+			);
+		};
+
+		const getLocationDetailsFromRequest = (
+			request: Request,
+			locationType?: string
+		) => {
+			const searchParams = new URLSearchParams( request.url() );
+
+			if ( locationType === 'product' ) {
+				return {
+					type: searchParams.get( 'location[type]' ),
+					productId: searchParams.get(
+						`location[sourceData][productId]`
+					),
+				};
+			}
+
+			if ( locationType === 'archive' ) {
+				return {
+					type: searchParams.get( 'location[type]' ),
+					taxonomy: searchParams.get(
+						`location[sourceData][taxonomy]`
+					),
+					termId: searchParams.get( `location[sourceData][termId]` ),
+				};
+			}
+
+			return {
+				type: searchParams.get( 'location[type]' ),
+				sourceData: searchParams.get( `location[sourceData]` ),
+			};
+		};
+
+		test( 'as product in specific Single Product template', async ( {
+			page,
+			pageObject,
+			editorUtils,
+		} ) => {
+			const productName = 'Cap';
+			const productSlug = 'cap';
+
+			await editorUtils.openSpecificProductTemplate(
+				productName,
+				productSlug
+			);
+
+			await editorUtils.insertBlockUsingGlobalInserter(
+				pageObject.BLOCK_NAME
+			);
+
+			await pageObject.chooseCollectionInTemplate( 'featured' );
+
+			const locationRequest = await page.waitForRequest( filterRequest );
+
+			const { type, productId } = getLocationDetailsFromRequest(
+				locationRequest,
+				'product'
+			);
```

Comment:
There might be something off here! If the `chooseCollectionInTemplate` is triggering the request we're waiting for, the order of execution should be the following:

```suggestion
			const locationRequestPromise = page.waitForRequest( filterRequest );

			await pageObject.chooseCollectionInTemplate( 'featured' );

			const { type, productId } = getLocationDetailsFromRequest(
				await locationRequestPromise,
				'product'
			);
```

This is because we want to be sure we catch the request, so we need to start waiting [before the request event is triggered](https://playwright.dev/docs/events#waiting-for-event).

Reply:
Thanks for all the hints! It's been addressed in https://github.com/woocommerce/woocommerce/pull/43997/commits/e53b15a6060ab0cdbd537816a2ebbc267cac04ad